src/logic/safe/hooks/useLoadSafe.tsx

```diff
@@ -13,6 +13,12 @@ export const useLoadSafe = (safeAddress?: string): boolean => {
   const dispatch = useDispatch<Dispatch>()
   const [isSafeLoaded, setIsSafeLoaded] = useState(false)
 
+  useEffect(() => {
+    if (isSafeLoaded) {
+      setIsSafeLoaded(false)
+    }
+  }, [safeAddress])
```

Comment:
It makes more sense now. I was indicating this potential problem in the other PR.